zipkin-collector/rabbitmq/src/test/java/zipkin2/collector/rabbitmq/ITRabbitMQCollector.java

```diff
@@ -15,151 +15,120 @@
-    try (RabbitMQCollector collector =
-        builder().addresses(Collections.singletonList(notRabbitMqAddress)).build()) {
-      thrown.expect(UncheckedIOException.class);
-      thrown.expectMessage("Unable to establish connection to RabbitMQ server: Connection refused");
-      collector.start();
-    }
+  @After public void after() throws Exception {
+    collector.close();
   }
 
   /** Ensures list encoding works: a json encoded list of spans */
-  @Test
-  public void messageWithMultipleSpans_json() throws Exception {
+  @Test public void messageWithMultipleSpans_json() throws Exception {
     messageWithMultipleSpans(SpanBytesEncoder.JSON_V1);
   }
 
   /** Ensures list encoding works: a version 2 json list of spans */
-  @Test
-  public void messageWithMultipleSpans_json2() throws Exception {
+  @Test public void messageWithMultipleSpans_json2() throws Exception {
     messageWithMultipleSpans(SpanBytesEncoder.JSON_V2);
   }
 
   /** Ensures list encoding works: proto3 ListOfSpans */
-  @Test
-  public void messageWithMultipleSpans_proto3() throws Exception {
+  @Test public void messageWithMultipleSpans_proto3() throws Exception {
     messageWithMultipleSpans(SpanBytesEncoder.PROTO3);
   }
 
   void messageWithMultipleSpans(SpanBytesEncoder encoder) throws Exception {
     byte[] message = encoder.encodeList(spans);
-    rabbit.publish(message);
+    publish(message);
 
-    Thread.sleep(1000);
-    assertThat(rabbit.storage.acceptedSpanCount()).isEqualTo(spans.size());
+    Thread.sleep(200L);
+    assertThat(storage.acceptedSpanCount()).isEqualTo(spans.size());
 
-    assertThat(rabbit.rabbitmqMetrics.messages()).isEqualTo(1);
-    assertThat(rabbit.rabbitmqMetrics.messagesDropped()).isZero();
-    assertThat(rabbit.rabbitmqMetrics.bytes()).isEqualTo(message.length);
-    assertThat(rabbit.rabbitmqMetrics.spans()).isEqualTo(spans.size());
-    assertThat(rabbit.rabbitmqMetrics.spansDropped()).isZero();
+    assertThat(rabbitmqMetrics.messages()).isEqualTo(1);
+    assertThat(rabbitmqMetrics.messagesDropped()).isZero();
+    assertThat(rabbitmqMetrics.bytes()).isEqualTo(message.length);
+    assertThat(rabbitmqMetrics.spans()).isEqualTo(spans.size());
+    assertThat(rabbitmqMetrics.spansDropped()).isZero();
   }
 
   /** Ensures malformed spans don't hang the collector */
-  @Test
-  public void skipsMalformedData() throws Exception {
+  @Test public void skipsMalformedData() throws Exception {
     byte[] malformed1 = "[\"='".getBytes(UTF_8); // screwed up json
     byte[] malformed2 = "malformed".getBytes(UTF_8);
-    rabbit.publish(THRIFT.encodeList(spans));
-    rabbit.publish(new byte[0]);
-    rabbit.publish(malformed1);
-    rabbit.publish(malformed2);
-    rabbit.publish(THRIFT.encodeList(spans));
+    publish(THRIFT.encodeList(spans));
+    publish(new byte[0]);
+    publish(malformed1);
+    publish(malformed2);
+    publish(THRIFT.encodeList(spans));
 
-    Thread.sleep(1000);
+    Thread.sleep(200L);
```

Comment:
As far as I know rabbit is push not poll, so any delay here would be how long it can decode the message and synchronously add to our in-memory storage. 200L is a bit excessive but less slow than 1s